std/pubkey.go

```diff
@@ -20,6 +20,9 @@ var _ types.PublicKeyCodec = DefaultPublicKeyCodec{}
 
 // Decode implements the PublicKeyCodec.Decode method
 func (cdc DefaultPublicKeyCodec) Decode(key *types.PublicKey) (crypto.PubKey, error) {
+	if key == nil {
+		return nil, nil
+	}
```

Comment:
if the pubkey is nil, `error` makes sense. Is there a reason to handle nil separately?

Reply:
nil is a valid pubkey, so it shouldn't error

Reply:
I thought PublicKey was a [sum](https://github.com/cosmos/cosmos-sdk/blob/f02b0b574501c81c65198ac3858a997ddc699f50/crypto/types/crypto.pb.go#L38). Can you explain how nil is a valid pubkey?

Reply:
I don't have full context, @aaronc can explain better

Reply:
All fields in proto are optional by default but pubkey is definitely optional. `nil` is correct here. I can't think of any other correct behavior. Let's add an explanatory comment.

Reply:
We should also add a case for `key.Sum` being `nil`.